conans/model/options.py

```diff
@@ -160,21 +163,38 @@ def __init__(self, values=None):
         if not values:
             return
 
+        assert not isinstance(values, six.string_types), "String type not expected in OptionValues " \
```

Comment:
Please use a if - elif structure to process all posible types.
I would allow string type, and would handle as single element tuple. It is reasonable that someone will write:
```python
default_options = "shared=True"
```

Reply:
`default_options` are parsed using [this function](https://github.com/conan-io/conan/blob/82631b05304f07dddfbd9f2cb0721e10fcd43d17/conans/model/conan_file.py#L18)  where the value `"shared=True"` will enter through `OptionsValues.loads(...)` so in `OptionsValues.__init__` we are never expecting a single string. We can refactor all this code, in fact I agree that it is more complex than it should be.